client/hooks/useAddPack.js

```diff
@@ -1,28 +1,59 @@
-// import fetcher from "../api/fetcher";
-// import { api } from "../constants/api";
-// import { useMutation } from "@tanstack/react-query";
-// import { queryClient } from "../constants/queryClient";
+import fetcher from "../api/fetcher";
+import { api } from "../constants/api";
+import { useMutation } from "@tanstack/react-query";
+import { queryClient } from "../constants/queryClient";
+import { trpc } from "../trpc";
 
-// const addPack = async (newPack) => {
-//   return await fetcher(`${api}/pack/`, {
-//     method: "POST",
-//     headers: {
-//       "Content-Type": "application/json",
-//     },
-//     body: JSON.stringify(newPack),
-//   });
-// };
 
-// export default function useAddPack() {
-//   const mutation = useMutation({
-//     mutationFn: async (newPack) => {
-//       return addPack(newPack);
-//     },
-//     onSuccess: () => {
-//       // Invalidate and refetch
-//       queryClient.invalidateQueries({ queryKey: ["packs"] });
-//     },
-//   });
+export default function useAddPack() {
+  const mutation = useMutation(
+    async (newPack) => {
+      try {
+        // Send the request to the server
+        const response = await trpc.addPack.mutate(newPack);
+        console.log("Response:", response);
+        return response;
+      } catch (error) {
+        throw error;
+      }
+    },
+    {
+      onMutate:async (newPack) => {
+         await queryClient.cancelQueries(["packs", newPack.owner_id])
+        // Step 1: Define optimistic update
+        const optimisticUpdate = {
+          ...newPack,
+          id: Date.now(),
+        };
+        console.log("Optimistic update")
+        console.log( newPack)
+        const oldQueryData= queryClient.getQueryData(["packs", newPack.owner_id]);
```

Comment:
Do we have to be using the actual queryClient here ? As opposed to the trpc instance you were using in others